oak_functions/loader/src/lookup.rs

```diff
@@ -79,59 +79,81 @@ impl LookupData {
     /// method, and existing entries are left untouched. The caller may retry the refresh operation
     /// at a future time.
     pub async fn refresh(&self) -> anyhow::Result<()> {
-        // TODO(#1930): Avoid loading the entire file in memory for parsing.
-        let https = HttpsConnector::with_native_roots();
-        let client = Client::builder().build::<_, Body>(https);
-        let start = Instant::now();
-        let lookup_data_buf = send_request(&client, self.build_download_request().await?).await?;
-        self.logger.log_public(
-            Level::Info,
-            &format!(
-                "downloaded {} bytes of lookup data in {:.0?}",
-                lookup_data_buf.len(),
-                start.elapsed()
-            ),
-        );
-
-        let start = Instant::now();
-        let entries = parse_lookup_entries(&mut lookup_data_buf.as_ref())
-            .context("could not parse lookup data")?;
-
-        self.logger.log_public(
-            Level::Info,
-            &format!(
-                "parsed {} entries of lookup data in {:.0?}",
-                entries.len(),
-                start.elapsed()
-            ),
-        );
-
-        // This block is here to emphasize and ensure that the write lock is only held for a very
-        // short time.
-        let start = Instant::now();
-        {
-            *self
-                .entries
-                .write()
-                .expect("could not lock entries for write") = entries;
+        match &self.lookup_data_source {
+            Some(lookup_data_source) => {
+                let start = Instant::now();
+                let lookup_data_buf: Bytes = match lookup_data_source {
```

Comment:
This method is doing too many things. I'd suggest moving the block for fetching `lookup_data_buf` into its own method, if possible.

Reply:
Done.